src/main.rs

```diff
@@ -1,244 +1,649 @@
+            );
+            let method_spec = metadata
+                .metadata_registration
+                .method_specs
+                .get(generic_class.generic_method_index as usize)
+                .unwrap();
+
+            cpp_context_collection.make_generic_from(method_spec, &mut metadata, &STATIC_CONFIG);
+        }
+    }
+    {
+        let total = metadata.metadata_registration.generic_method_table.len() as f64;
+        println!("Filling generic types!");
+        for (i, generic_class) in metadata
+            .metadata_registration
+            .generic_method_table
+            .iter()
+            .enumerate()
+        {
+            println!(
+                "Filling generic type instantiations {:.4}% ({i}/{total})",
+                (i as f64 / total * 100.0)
+            );
+            let method_spec = metadata
+                .metadata_registration
+                .method_specs
+                .get(generic_class.generic_method_index as usize)
+                .unwrap();
+
+            cpp_context_collection.fill_generic_class_inst(
+                method_spec,
+                &mut metadata,
+                &STATIC_CONFIG,
+            );
+        }
+    }
+
+    if cli.gen_generic_methods_specializations {
+        let total = metadata.metadata_registration.generic_method_table.len() as f64;
+        println!("Filling generic methods!");
+        for (i, generic_class) in metadata
+            .metadata_registration
+            .generic_method_table
+            .iter()
+            .enumerate()
+        {
+            println!(
+                "Filling generic method instantiations {:.4}% ({i}/{total})",
+                (i as f64 / total * 100.0)
+            );
+            let method_spec = metadata
+                .metadata_registration
+                .method_specs
+                .get(generic_class.generic_method_index as usize)
+                .unwrap();
+
+            cpp_context_collection.fill_generic_method_inst(
+                method_spec,
+                &mut metadata,
+                &STATIC_CONFIG,
+            );
+        }
+    }
+
+    println!("Registering handlers!");
+    unity::register_unity(&mut metadata)?;
+    value_type::register_value_type(&mut metadata)?;
+    println!("Handlers registered!");
+
+    {
+        // Fill them now
+        println!("Filling types");
+        let type_defs = metadata.metadata.global_metadata.type_definitions.as_vec();
+        let total = type_defs.len();
+        for tdi_u64 in 0..total {
+            let tdi = TypeDefinitionIndex::new(tdi_u64 as u32);
+
+            println!(
+                "Filling type {:.4} ({tdi_u64}/{total})",
+                (tdi_u64 as f64 / total as f64 * 100.0)
+            );
+
+            cpp_context_collection.fill(
+                &metadata,
+                &STATIC_CONFIG,
+                CppTypeTag::TypeDefinitionIndex(tdi),
+            );
+        }
+    }
+
+    let write_all = false;
```

Comment:
This will mean cordl built from source will not generate all headers for us, this should be set to true probably